src/frontend/src/flows/authorize/fetchDelegation.ts

```diff
@@ -5,28 +5,31 @@ import {
 import { toast } from "$src/components/toast";
 import { AuthenticatedConnection } from "$src/utils/iiConnection";
 import { unknownToString } from "$src/utils/utils";
+import { Signature } from "@dfinity/agent";
 import { nonNullish } from "@dfinity/utils";
-import { AuthContext, Delegation } from "./postMessageInterface";
+import { Delegation } from "./postMessageInterface";
 
 /**
- * Prepares and fetches a delegation valid for the authenticated user and the application information contained in
- * authContext.
- * @param userNumber User number resulting from successful authentication.
+ * Prepares and fetches a delegation valid for the authenticated user and the derivation.
  * @param connection authenticated II connection resulting from successful authentication.
- * @param authContext Information about the authentication request received from the application via window post message.
+ * @param derivationOrigin the origin for which to create the delegation
+ * @param publicKey the key to delegate to
+ * @param maxTimeToLive until when the delegation is valid (nanoseconds since now)
  * @return Tuple of PublicKey and matching delegation.
  */
-export const fetchDelegation = async (
-  userNumber: bigint,
-  connection: AuthenticatedConnection,
-  authContext: AuthContext
-): Promise<[PublicKey, Delegation] | { error: unknown }> => {
-  const sessionKey = Array.from(authContext.authRequest.sessionPublicKey);
-
+export const fetchDelegation = async ({
+  connection,
+  derivationOrigin: derivationOrigin_,
+  publicKey,
+  maxTimeToLive,
+}: {
+  connection: AuthenticatedConnection;
+  derivationOrigin: string;
+  publicKey: Uint8Array;
+  maxTimeToLive?: bigint;
+}): Promise<[PublicKey, Delegation] | { error: unknown }> => {
   // at this point, derivationOrigin is either validated or undefined
```

Comment:
Maybe move this comment to `src/frontend/src/flows/authorize/postMessageInterface.ts:158` as well?